### ISSUE TITLE
swagger-tools breaks this document

### ISSUE THREAD

**Issue by mohsen1** [apigee-127/swagger-tools#180]
[This document](https://github.com/swagger-api/swagger-editor/blob/95e6616e901f5c0b99af402cd0fd27fb59775f81/app/spec-files/petstore_simple.yaml) breaks the validation mechanism 

I'm guessing it's because of [this](https://github.com/swagger-api/swagger-editor/blob/95e6616e901f5c0b99af402cd0fd27fb59775f81/app/spec-files/petstore_simple.yaml#L134-L144) section where it uses `allOf` and `$ref`

```
swagger-tools validate petstore_simple.yaml

Error: ptr is required
    at Object.pathFromPointer (/usr/local/lib/node_modules/swagger-tools/node_modules/json-refs/index.js:189:11)
    at /usr/local/lib/node_modules/swagger-tools/lib/specs.js:401:44
    at arrayEach (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:1290:13)
    at Function.forEach (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:5952:11)
    at /usr/local/lib/node_modules/swagger-tools/lib/specs.js:387:11
    at baseFor (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:2007:13)
    at baseForOwn (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:2061:14)
    at baseEach (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:1820:16)
    at Function.forEach (/usr/local/lib/node_modules/swagger-tools/node_modules/lodash/index.js:5953:11)
    at processDocument (/usr/local/lib/node_modules/swagger-tools/lib/specs.js:352:5)
    at validateSwagger2_0 (/usr/local/lib/node_modules/swagger-tools/lib/specs.js:883:3)
```

**Comment by mohsen1** [comment]
BTW it **is** that section. I was able to go around the issue by [this](https://github.com/swagger-api/swagger-editor/commit/7d5bfac371f6b1fe5fc3634ac7fefc435831393e) commit

**Comment by whitlockjc** [comment]
I am unable to reproduce this.  When I run `swagger-tools validate petstore_simple.yaml`, I get this:

```
API Warnings:

  #/definitions/newPet/allOf/1: Definition is defined but is not used: #/definitions/newPet/allOf/1

Validation succeeded but with 1 warning
```

Since I cannot reproduce this, I am going to close this and file the bug above that shows me inline models do not always get counted as self-referenced and can result in being reported as unused.

**Comment by mohsen1** [comment]
at which version/sha1 you confirmed this is not a bug?

**Comment by whitlockjc** [comment]
master

**Comment by mohsen1** [comment]
Master is moving so I don't know which sha exactly. Also when it's released it should be marked as fixed. I

**Comment by whitlockjc** [comment]
I could reproduce this with 0.8.5 but with master, this does not happen so since this is fixed in master and will be out in the next release, it made sense to close it.

**Comment by whitlockjc** [comment]
Just so it doesn't seem like magic, this was fixed here: fb2f262.

**Comment by mohsen1** [comment]
Thanks for making it clear 👍
